Change to use blocknumber instead of timestamp in withdrawal status

rlgns98kr opened the pull request:
## Motivation
The `l1Timestamp` of `L2OutputOracle`'s `OutputProposed` Event indicates the time when the output was submitted in L1. Comparing this with the timestamp of the L2 block may result in an incorrect comparison.

For example,
when a withdrawal tx occurred 3 minutes ago in block 14 in L2, the most recent output submitted to the L2OutputOracle Contract is 1 minute ago, and the l2 block number in the most recent output is 10, timestamp comparison is meaningless. (This may occur due to gas price, etc.)

before PR, https://github.com/blockscout/blockscout/pull/8702

## Checklist for your Pull Request (PR)

  - [x] I added an entry to `CHANGELOG.md` with this PR
  - [ ] If I added new functionality, I added tests covering it.
  - [ ] If I fixed a bug, I added a regression test to prevent the bug from silently reappearing again.
  - [x] I checked whether I should update the docs and did so by submitting a PR to https://github.com/blockscout/docs
  - [ ] If I added/changed/removed ENV var, I submitted a PR to https://github.com/blockscout/docs to update the list of env vars at https://github.com/blockscout/docs/blob/master/for-developers/information-and-settings/env-variables.md and I updated the version to `master` in the Version column. Changes will be reflected in this table: https://docs.blockscout.com/for-developers/information-and-settings/env-variables. 
  - [ ] If I add new indices into DB, I checked, that they are not redundant with PGHero or other tools

rlgns98kr commented:
@varasev
Thank you for your checking. I opened the new PR now. 🙇🏼‍♂️

varasev commented on apps/explorer/lib/explorer/chain.ex:
```diff
@@ -1889,7 +1889,7 @@ defmodule Explorer.Chain do
         select: %{
           hash: w.hash,
-          l2_timestamp: l2_block.timestamp,
+          l2_block_number: l2_block.number,
```
I agree with the changes, but let's replace `l2_block.number` with `w.l2_block_number` here. It will be more reliable taking into account that we `left join` the block info that may be missing at the moment of query (i.e. `l2_block` can be empty).

rlgns98kr replied:
Oh, I understood it.
Thank you for your fast and valuable feedback and for correcting my carelessness. 😊
Let me update it next Monday. I hope you have a happy weekend.

rlgns98kr commented:
I replaced `l2_block.number` with `w.l2_block_number` and tested it in local and our team's production! I think it works well.

Thank you! 🙇🏼‍♂️
I'm sorry for my forced update. I fixed my typo in my commit (tiemstamp -> timestamp).

rlgns98kr commented:
@vbaranov 
Hello?
Sorry for the sudden tag.
I tagged you because you are the biggest contributor to this repository.

I really want this change to be applied to the optimism-production branch. Is there anything else I need to do? I will do my best.
